### ISSUE TITLE
Issue with role dependency variable substitution and include task in Ansible 2.1.0

### ISSUE THREAD

**Issue by mickael-ange** [ansible/ansible#16080]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
$ ansible-playbook --version
ansible-playbook 2.1.0.0
  config file = /home/mickael/git/ansible-reports/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

n/a
##### OS / ENVIRONMENT

Ansible controler: Ubuntu 14.04
Managed platform: Ubuntu 14.04 and CentOS7
##### SUMMARY

Variable in an include task of a dependent role has unexpected value in 2.1.0 compared to Ansible 2.0.2.0.
##### STEPS TO REPRODUCE

Create a playbook.
Add a role (test-vars-01) with a role (test-vars-01-dep1) dependency.
Override a defaults var of `test-vars-01` role and pass the variable from the playbook to the `test-vars-01` role.
Then pass the var to the dependency using `test-vars-01/meta/main.yml`.
Display the var value (using debug module) in an include task in `test-vars-01-dep1` role.
-> var has the default value defined in `test-vars-01/defaults/main.yml` instead of the override value passed from the playbook.
- Directory structure

```
├── ansible.cfg
├── group_vars
│   └── all
│       └── test-vars-01.yml
├── hosts
├── playbooks
│   └── test-vars-01.yml
├── README.md
└── roles
    ├── test-vars-01
    │   ├── defaults
    │   │   └── main.yml
    │   ├── meta
    │   │   └── main.yml
    │   └── tasks
    │       ├── debug.yml
    │       └── main.yml
    └── test-vars-01-dep1
        ├── defaults
        │   └── main.yml
        └── tasks
            ├── debug.yml
            └── main.yml
```
- Group Variables

`group_vars/all/test-vars-01.yml`

```

---
group_vars_myvar: group_vars_value
```
- Playbook

`playbooks/test-vars-01.yml`

```

---
- hosts: localhost
  roles:
    - role: test-vars-01
      test_myvar: "{{ group_vars_myvar }}"
  tasks:
    - debug:
        msg: "test_myvar={{ test_myvar }} (after role played)"
```
- `test-vars-01` role

`roles/test-vars-01/defaults/main.yml`

```

---
test_myvar: defaults
```

`roles/test-vars-01/meta/main.yml`

```

---
dependencies:
  - role: test-vars-01-dep1
    test_dep1_myvar: "{{ test_myvar }}"
```

`roles/test-vars-01/tasks/main.yml`

```

---
- debug:
    msg: "test_myvar={{ test_myvar }} (from roles/test-vars-01/tasks/main.yml)"

- include: debug.yml
```

`roles/test-vars-01/tasks/debug.yml`

```

---
- debug:
    msg: "test_myvar={{ test_myvar }} (from roles/test-vars-01/tasks/debug.yml)"
```
- `test-vars-01-dep1` role

`roles/test-vars-01-dep1/defaults/main.yml`

```

---
test_dep1_myvar: defaults-dep1
```

`roles/test-vars-01-dep1/tasks/main.yml`

```

---
- debug:
    msg: "test_dep1_myvar={{ test_dep1_myvar }} (from roles/test-vars-01-dep1/tasks/main.yml)"

- include: debug.yml
```

`roles/test-vars-01-dep1/tasks/debug.yml`

```

---
- debug:
    msg: "test_dep1_myvar={{ test_dep1_myvar }} (from included task roles/test-vars-01-dep1/tasks/debug.yml)"
```

Everything is available on Github: https://github.com/mickael-ange/ansible-reports
##### EXPECTED RESULTS

With Ansible 2.0.2.0, I have the expected result. The value `group_vars_value` is propagated to dependent roles var `test_dep1_myvar`, as well as in the included tasks.

```
TASK [test-vars-01-dep1 : debug] ***********************************************
ok: [localhost] => {
    "msg": "test_dep1_myvar=group_vars_value (from included task roles/test-vars-01-dep1/tasks/debug.yml)"
}
```

```
$ ansible-playbook --version
ansible-playbook 2.0.2.0
  config file = /home/mickael/git/ansible-reports/ansible.cfg
  configured module search path = Default w/o overrides
```

```
$ ansible-playbook playbooks/test-vars-01.yml 

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [test-vars-01-dep1 : debug] ***********************************************
ok: [localhost] => {
    "msg": "test_dep1_myvar=group_vars_value (from roles/test-vars-01-dep1/tasks/main.yml)"
}

TASK [test-vars-01-dep1 : include] *********************************************
included: /home/mickael/git/ansible-reports/roles/test-vars-01-dep1/tasks/debug.yml for localhost

TASK [test-vars-01-dep1 : debug] ***********************************************
ok: [localhost] => {
    "msg": "test_dep1_myvar=group_vars_value (from included task roles/test-vars-01-dep1/tasks/debug.yml)"
}

TASK [test-vars-01 : debug] ****************************************************
ok: [localhost] => {
    "msg": "test_myvar=group_vars_value (from roles/test-vars-01/tasks/main.yml)"
}

TASK [test-vars-01 : include] **************************************************
included: /home/mickael/git/ansible-reports/roles/test-vars-01/tasks/debug.yml for localhost

TASK [test-vars-01 : debug] ****************************************************
ok: [localhost] => {
    "msg": "test_myvar=group_vars_value (from roles/test-vars-01/tasks/debug.yml)"
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "test_myvar=defaults (after role played)"
}

PLAY RECAP *********************************************************************
localhost                  : ok=8    changed=0    unreachable=0    failed=0   
```
##### ACTUAL RESULTS

With Ansible 2.1.0, I have an _unexpected_ result. In the role `test-vars-01-dep1` (dependency), the var `test_dep1_myvar=group_vars_value` in `tasks/main.yml` but in the following include task the value becomes `test_dep1_myvar=defaults` (expected to be the same value `test_dep1_myvar=group_vars_value` as per Ansible 2.0.2.0). 

```
TASK [test-vars-01-dep1 : debug] ***********************************************
task path: /home/mickael/git/ansible-reports/roles/test-vars-01-dep1/tasks/debug.yml:2
ok: [localhost] => {
    "msg": "test_dep1_myvar=defaults (from included task roles/test-vars-01-dep1/tasks/debug.yml)"
}
```

```
$ ansible-playbook --version
ansible-playbook 2.1.0.0
  config file = /home/mickael/git/ansible-reports/ansible.cfg
  configured module search path = Default w/o overrides
```

```
$ ansible-playbook -vvvv playbooks/test-vars-01.yml 
Using /home/mickael/git/ansible-reports/ansible.cfg as config file
Loaded callback default of type stdout, v2.0

PLAYBOOK: test-vars-01.yml *****************************************************
1 plays in playbooks/test-vars-01.yml

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: mickael
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1464766862.58-278987971600953 `" && echo ansible-tmp-1464766862.58-278987971600953="` echo $HOME/.ansible/tmp/ansible-tmp-1464766862.58-278987971600953 `" ) && sleep 0'
<localhost> PUT /tmp/tmp3Rv4i2 TO /home/mickael/.ansible/tmp/ansible-tmp-1464766862.58-278987971600953/setup
<localhost> EXEC /bin/sh -c 'LANG=C LC_ALL=C LC_MESSAGES=C /usr/bin/python /home/mickael/.ansible/tmp/ansible-tmp-1464766862.58-278987971600953/setup; rm -rf "/home/mickael/.ansible/tmp/ansible-tmp-1464766862.58-278987971600953/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]

TASK [test-vars-01-dep1 : debug] ***********************************************
task path: /home/mickael/git/ansible-reports/roles/test-vars-01-dep1/tasks/main.yml:2
ok: [localhost] => {
    "msg": "test_dep1_myvar=group_vars_value (from roles/test-vars-01-dep1/tasks/main.yml)"
}

TASK [test-vars-01-dep1 : include] *********************************************
task path: /home/mickael/git/ansible-reports/roles/test-vars-01-dep1/tasks/main.yml:5
included: /home/mickael/git/ansible-reports/roles/test-vars-01-dep1/tasks/debug.yml for localhost

TASK [test-vars-01-dep1 : debug] ***********************************************
task path: /home/mickael/git/ansible-reports/roles/test-vars-01-dep1/tasks/debug.yml:2
ok: [localhost] => {
    "msg": "test_dep1_myvar=defaults (from included task roles/test-vars-01-dep1/tasks/debug.yml)"
}

TASK [test-vars-01 : debug] ****************************************************
task path: /home/mickael/git/ansible-reports/roles/test-vars-01/tasks/main.yml:2
ok: [localhost] => {
    "msg": "test_myvar=group_vars_value (from roles/test-vars-01/tasks/main.yml)"
}

TASK [test-vars-01 : include] **************************************************
task path: /home/mickael/git/ansible-reports/roles/test-vars-01/tasks/main.yml:5
included: /home/mickael/git/ansible-reports/roles/test-vars-01/tasks/debug.yml for localhost

TASK [test-vars-01 : debug] ****************************************************
task path: /home/mickael/git/ansible-reports/roles/test-vars-01/tasks/debug.yml:2
ok: [localhost] => {
    "msg": "test_myvar=group_vars_value (from roles/test-vars-01/tasks/debug.yml)"
}

TASK [debug] *******************************************************************
task path: /home/mickael/git/ansible-reports/playbooks/test-vars-01.yml:7
ok: [localhost] => {
    "msg": "test_myvar=defaults (after role played)"
}

PLAY RECAP *********************************************************************
localhost                  : ok=8    changed=0    unreachable=0    failed=0 
```

**Comment by kal4l** [comment]
I am experiencing this exact same issue on 2.0.0.2.

```
$ ansible --version
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = /usr/share/ansible
```

**Comment by jimi-c** [comment]
Per my comment here: https://github.com/ansible/ansible/issues/16320#issuecomment-227912789, this issue appears to be resolved and will be included in 2.1.1.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by mickael-ange** [comment]
Just try to upgrade to Ansible 2.1.1.0 and I can confirm that the bug is fixed.
Thank you.
